includes/Checker/Checks/Abstract_File_Check.php

```diff
@@ -189,11 +188,11 @@ private static function file_get_contents( $file ) {
 	 *
 	 * @since n.e.x.t
 	 *
-	 * @param Check_Context $plugin Context for the plugin to check.
+	 * @param Check_Result $result The Check Result to amend.
 	 * @return array List of absolute file paths.
 	 */
-	private static function get_files( Check_Context $plugin ) {
-		$location = $plugin->location();
+	private static function get_files( Check_Result $result ) {
+		$location = $result->plugin()->location();
```

Comment:
I don't think these changes are needed for the logic you're introducing here? `Plugin_Request_Utility::is_single_file_plugin()` only looks at `$result->plugin()`, so it would be better to keep this as is and only pass the `Check_Context $plugin` to both this method and `Plugin_Request_Utility::is_single_file_plugin()`.

Methods should have as much access as needed, but as little access as possible.

Reply:
Got it. Reverted in https://github.com/WordPress/plugin-check/pull/319/commits/6c6c3dfadd99217bec53e3536d255c0087cc86bf